### PR TITLE
Added middleware to catch uncaught errors

### PR DESCRIPTION
**Pull request by carlosascari**
Determines if the error thrown because the api server is down
Also determines if api server is up, but fetching failed
Any other error is logged

### CODE REVIEW

**Comment by asood123** [comment]
Thanks for the quick response. It's breaking the 404 page in tests. Adding @arnaudbenard as well

File: server/middlewares.js [hunk 1]
```diff
@@ -137,6 +137,38 @@ const addTitle = (title) => {
     next();
   }
 }
+
+const handleUncaughtError = (error, req, res, next) => {
+    if (error.name === 'FetchError')
+    {
```
Comment (asood123) [inline server/middlewares.js]: We usually put braces on the previous line. http://eslint.org/docs/rules/brace-style#tbs

Comment (carlosascari) [inline server/middlewares.js]: fixed :+1:

**Comment by carlosascari** [comment]
@asood123, @sedubois, I'm not sure what happened here

> Your build ran 168 tests in junit with 0 failures

...

> Running:  Redirects to Github after clicking connect
TEST FAILURE: 1 assertions failed, 59 passed (55.958s)
 ✖ connect_account_page
   - Redirects to Github after clicking connect
     Testing if the URL contains "http://localhost:3000/testcollective"

**Comment by arnaudbenard** [comment]
When I run `website` without `api`, I get the standard error message. Looks like the proper fix 👍

**Pull request merged**